File: library/src/main/java/io/ashdavies/rxtasks/TaskListenerFactory.java
package io.ashdavies.rxtasks;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;

interface TaskListenerFactory<Result, Emitter> {

  OnSuccessListener<Result> createOnSuccessListener(Emitter emitter);

  OnFailureListener createOnFailureListener(Emitter emitter);
}

File: src/main/kotlin/io/ashdavies/rx/rxtasks/TaskListenerFactory.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.OnFailureListener
import com.google.android.gms.tasks.OnSuccessListener

internal interface TaskListenerFactory<T, in E> {

  fun createOnSuccessListener(emitter: E): OnSuccessListener<T>

  fun createOnFailureListener(emitter: E): OnFailureListener
}
